pkg/build/os-release.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Chainguard, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"chainguard.dev/apko/pkg/build/types"
+	"chainguard.dev/apko/pkg/options"
+)
+
+func (di *defaultBuildImplementation) GenerateOSRelease(
+	o *options.Options, ic *types.ImageConfiguration,
+) error {
+	path := filepath.Join(o.WorkDir, "etc", "os-release")
+
+	_, err := os.Stat(path)
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+
+	w, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer w.Close()
+
+	if ic.OSRelease.ID != "" {
+		_, err := fmt.Fprintf(w, "ID=%s\n", ic.OSRelease.ID)
+		if err != nil {
+			return err
+		}
+	}
+
+	if ic.OSRelease.Name != "" {
+		_, err := fmt.Fprintf(w, "NAME=\"%s\"\n", ic.OSRelease.Name)
+		if err != nil {
+			return err
+		}
+	}
+
+	if ic.OSRelease.PrettyName != "" {
+		_, err := fmt.Fprintf(w, "PRETTY_NAME=\"%s\"\n", ic.OSRelease.PrettyName)
+		if err != nil {
+			return err
+		}
+	}
+
+	if ic.OSRelease.VersionID != "" {
+		_, err := fmt.Fprintf(w, "VERSION_ID=%s\n", ic.OSRelease.VersionID)
+		if err != nil {
+			return err
+		}
+	}
+
+	if ic.OSRelease.HomeURL != "" {
+		_, err := fmt.Fprintf(w, "HOME_URL=\"%s\"\n", ic.OSRelease.HomeURL)
+		if err != nil {
+			return err
+		}
+	}
+
+	if ic.OSRelease.BugReportURL != "" {
+		_, err := fmt.Fprintf(w, "BUG_REPORT_URL=\"%s\"\n", ic.OSRelease.BugReportURL)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
What do you think if we do this in a template later to skip all the ifs and error checks?

Reply:
sure